### PR TITLE
Editor typog drawer

### PR DESCRIPTION
**Pull request by csiyang**
Added drawers for attributes in typography block
Drawer toggles can update backend data

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/TextColor/TextColor.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  return (
+    <ToggleButtonGroup
+      orientation="vertical"
+      value={selected}
+      exclusive
+      onChange={handleChange}
+      fullWidth
+      sx={{
+        display: 'flex',
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/TextColor/TextColor.tsx:41]: Minor but you could put all these styles on 1 line? Makes the file shorter so less to scroll through when quickly reading

File: apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  TypographyAlign,
+  TypographyVariant,
+  ThemeMode,
+  ThemeName
+} from '../../../../../__generated__/globalTypes'
+import { GetJourney_journey_blocks_StepBlock as StepBlock } from '../../../../../__generated__/GetJourney'
+import { ColorDisplayIcon } from '.'
+
+describe('ColorDisplayIcon', () => {
+  it('should show the color', () => {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.spec.tsx:15]:
```suggestion
  it('should show the selected color', () => {
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ) as TreeBlock<CardBlock> | undefined
+
+  return (
+    <Paper
+      sx={{
+        borderRadius: 1000
+      }}
+    >
+      <ThemeProvider
+        themeName={card?.themeName ?? ThemeName.base}
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.tsx:35]: Use the journey theme?
```suggestion
        themeName={card?.themeName ?? journey.themeName}
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        themeMode={card?.themeMode ?? ThemeMode.dark}
+        nested
+      >
+        <Box
+          data-testid="color-display-icon"
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/ColorDisplayIcon/ColorDisplayIcon.tsx:40]:
```suggestion
          data-testid=`${color}-color-icon`
```
Make sure id's are unique

File: apps/journeys-admin/src/components/ThemeProvider/admin/tokens/components.ts [hunk 5]
```diff
@@ -44,6 +44,39 @@ export const adminComponents: Required<Pick<ThemeOptions, 'components'>> = {
           borderColor: palette[200]
         }
       }
+    },
+    MuiToggleButtonGroup: {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/ThemeProvider/admin/tokens/components.ts:48]: This might need to be moved in the future..

**Review by storyworks — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/FontVariant/FontVariant.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { render } from '@testing-library/react'
+import { FontVariant } from '.'
+
+describe('TextVariant drawer', () => {
+  it('should show default values', () => {
+    const { getByText, getByRole } = render(
+      <FontVariant id={'font-variant-id'} variant={null} />
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/FontVariant/FontVariant.spec.tsx:7]: Renaming to Variant as discussed

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/TextAlign/TextAlign.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    $input: TypographyBlockUpdateInput!
+  ) {
+    typographyBlockUpdate(id: $id, journeyId: $journeyId, input: $input) {
+      id
+      align
+    }
+  }
+`
+
+export function TextAlign({ id, align }: TextAlignProps): ReactElement {
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/TextAlign/TextAlign.tsx:32]: renamed to Align as discussed

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Typography.tsx [hunk 8]
```diff
@@ -1,64 +1,71 @@
...
         name="Font Variant"
         value={capitalize(
           lowerCase(variant?.toString() ?? 'body2').replace('h', 'header')
         )}
         description="Font Variant"
-        // onClick open drawer
+        onClick={() => {
+          dispatch({
+            type: 'SetDrawerPropsAction',
+            title: 'Font Variant',
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Typography.tsx:48]:
```suggestion
            title: 'Text Variant',
```

And I've asked Vlad and Ariel to change design

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Variant/Variant.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { render, fireEvent, waitFor } from '@testing-library/react'
+import { Variant } from '.'
+
+describe('Typography variant selector', () => {
+  it('should show variant properties', () => {
+    const { getByRole } = render(
+      <Variant id={'font-variant-id'} variant={null} />
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Variant/Variant.spec.tsx:7]:
```suggestion
      <Variant id={'text-variant-id'} variant={null} />
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Variant/Variant.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    'caption',
+    'overline'
+  ]
+  const sorted = Object.values(TypographyVariant).sort(
+    (a, b) => order.indexOf(a) - order.indexOf(b)
+  )
+
+  function handleChange(
+    event: React.MouseEvent<HTMLElement>,
+    varint: TypographyVariant
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Typography/Variant/Variant.tsx:39]:
```suggestion
    variant: TypographyVariant
```

**Comment by storyworks** [comment]
Found a bug when running the app locally

> Go to the card with the typography block and RadioQuestion,
> Click on typography and change header variant
> Click onto RadioQuestion - header selectors still exist in drawer
> Click through header selectors - they change even though RadioQuestion is selected

Similar for switching between different types of typography blocks. We should update the selectors for each block selected..

Seeing it all working together is so satisfying though - well done! :D

**Review by storyworks — APPROVED** [review]

**Pull request merged**